### ISSUE TITLE
Reverse proxy?

### ISSUE THREAD

**Issue by gerroon** [prologic/tube#8]
Hi

Can you please provide reverse proxy examples for Apache/Nginx?

thanks

**Comment by prologic** [comment]
Yes I will provide one :) Although I use neither I'll put some example working configs in the repo!

**Comment by prologic** [comment]
Relevant documentation on this:

- https://docs.nginx.com/nginx/admin-guide/web-server/reverse-proxy/#proxy_bind
- https://httpd.apache.org/docs/2.4/mod/mod_proxy.html#examples

**Comment by gerroon** [comment]
The question is this. Doies it work behind reverse proxy if so what are the actual parameters. Some sites require additional settings for sockets etc. Also does it need iots own subdomain whgen it is proxied?

**Comment by prologic** [comment]
> The question is this. Does it work behind reverse proxy?

Yes! I currently deploy a production instance behind [Traefik](https://containo.us/traefik/) [here](https://tube.mills.io). You can find its configuration (_or very simialr_) in the [docker-compose.yml](/docker-compose.yml).

> if so what are the actual parameters. Some sites require additional settings for sockets etc.

No, nothing special.

> Also does it need its own subdomain when it is proxied?

Yes it currently does because some of the static assets are assumed to be available at absolute URIs and use `/static/...` for example. This _could_ be fixed I suppose but it makes it a bit more complicated.

**Comment by gerroon** [comment]
Thanks.
It would be nice  if the subdomain requirement is not necessity. The reason is that when you runa site under the subdomain, suddently your site is discoverable, if you will. I do not want people just watching my shares or whatever on the site.

**Comment by prologic** [comment]
Actually; I don't understand the premise of your statement :) Can you help me understand how a sub-domain is more discoverable than a path? Both are discoverable AFAIK and even crawlable by web crawlers unless you setup a `/robots.txt` (_assuming crawlers respect it at all_).

Do you mind filing another issue for making Tube work on a sub-path? I'll work on that today.

Also you actually seem to have quite a bit of experience with both Apache/NGINX Reverse Proxy setups so I would actually appreciate it if you put up a PR with example configs. Say in `./contrib/`? I would myself but I'm a bit out of touch with the traditional Reverse Proxy LB (s).

**Comment by gerroon** [comment]
When you have a subdomain like `myvideos.mydomain.com` they can just access the dns records. Therefor it is out there.

How can they know `mydomain.com/myvideos` exists if I never linked it in anywhere on th web and `mydomain.com` root directs to just 404?

**Comment by prologic** [comment]
> When you have a subdomain like `myvideos.mydomain.com` they can just access the dns records. Therefor it is out there.

Ahh I did not think of this :)

I actually run my own internal DNS so this isn't a problem for me :)

> How can they know `mydomain.com/myvideos` exists if I never linked it in anywhere on th web and `mydomain.com` root directs to just 404?

Fair point :)

> I would like to test the reverse proxy but I wont be setting it up myself since it requires a subdomain. I would be happy to test this out when it is possible to run it under a sub path.

=> #11

**Comment by prologic** [comment]
No we will still do this. I have unfortunately very little bandwidth myself right now due to my day job. If the community can help out here that would be awesome!

**Comment by rdd2** [comment]
Hi everyone, 
after much trial-and-error I managed to find an nginx-based workaround for the subdirectory problem. I have the following location working for tube:
```
location /subdirectory/ {
            proxy_set_header Host $http_host;
            proxy_pass http://tubehost:8000/;
            proxy_redirect / /subdirectory/;
            
            sub_filter 'href="/'  'href="/subdirectory/';
            sub_filter 'src="/'  'src="/subdirectory/';
            sub_filter_once off;
}
```
So I'm proxying and rewriting and then using sub_filter to change the links in the returned html to match the subdirectory. Pretty hacky, but it works and I hope this helps someone. 

Thomas

**Comment by prologic** [comment]
This needs a `$BASE_URL` or `--base-url` configuration like many of my other [Self-Hosted Apps](https://gist.github.com/prologic/c165403fb6ac07c16dc9fd6eeb8efbd2); if someone were to put up a PR for this I would happily review and accept it. It would eliminate the need for such complex reverse proxy setups :)

**Comment by prologic** [comment]
on hold, pr(s) welcome
